serve-api/handlers/order_test.go

```diff
@@ -0,0 +1,171 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"github.com/ministryofjustice/serve-opg/serve-api/entity"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+// MockOrderRepository is a mock type for model.UserRepository
+type MockOrderRepository struct {
+	mock.Mock
+}
+
+// SelectOrderByID is mock of OrderRepository SelectOrderByID
+func (m *MockOrderRepository) SelectOrderByID(id int) (*entity.Order, error) {
+	args := m.Called(id)
+
+	var order entity.Order
+	if args.Get(0) != nil {
+		order = args.Get(0).(entity.Order)
+	}
+
+	var e error
+
+	if args.Get(1) != nil {
+		e = args.Get(1).(error)
+	}
+
+	return &order, e
+}
+
+func (m *MockOrderRepository) GetServedOrders(dateLimit ...time.Time) ([]entity.Order, error) {
+	args := m.Called()
+
+	var orders []entity.Order
+	if args.Get(0) != nil {
+		orders = args.Get(0).([]entity.Order)
+	}
+
+	var e error
+
+	if args.Get(1) != nil {
+		e = args.Get(1).(error)
+	}
+
+	return orders, e
+}
+
+func (m *MockOrderRepository) TableName() string {
+	return "dc_order"
+}
+
+func TestSelectOrderByID(t *testing.T) {
+	order := entity.Order{
+		ID:          2,
```

Comment:
I know this probably won't affect the test but I guess if we're pretending to get an Order based on an ID then the order id should match the request?